src/components/ComboBox/ComboBox.js

```diff
@@ -175,6 +199,13 @@ export default class ComboBox extends React.Component {
     );
   };
 
+  onToggleClick = isOpen => event => {
+    if (event.target === this.textInput.current && isOpen) {
```

Comment:
Is there a way we could do this without dropping down into downshift implementation details and without persisting the event?

Reply:
I tried a couple of other approaches before getting to this one because none of them worked as good as this one.
The first one and the most obvious was to open the menu in the input's onFocus event. It worked with a timeout, which doesn't look good because it makes a weird jump.
Also, tried using Downshift's stateReducer prop, but there is no way to know if the state change comes from the input or the button or the clear selection.

Please let me know if you have any suggestion I can try/implement to avoid persisting the event and getting into Downshift details.